src/engine_render.c

```diff
@@ -5363,9 +5363,70 @@ static void display_fast_drawlist(struct Camera *cam)
     }
 }
 
-static long convert_world_coord_to_front_view_screen_coord(struct Coord3d* pos, struct Camera* cam, long* x, long* y, long* z)
+// Returns 1 if projected point is withing player's window, 0 otherwise
+static long project_point_helper(struct PlayerInfo *player, int zoom, int a1, int a2, int a3, unsigned short pos_z, long *x_out, long *y_out, long *z_out)
 {
-    return _DK_convert_world_coord_to_front_view_screen_coord(pos, cam, x, y, z);
+    int v11;
+    int64_t v13;
+    uint8_t v13_byte4;
+    short window_width = player->engine_window_width;
+    short window_height = player->engine_window_height;
+
+    *x_out = (zoom * a2 >> 16) + (*(uint16_t *)&window_width >> 1);
```

Comment:
Lots of typecasting and bit-shifting here (and below) does not make for readable or easy to understand code.